sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -216,6 +231,18 @@ public void configureGuice(DruidInjectorBuilder builder)
     {
     }
 
+
+    public File newTempFolder()
```

Comment:
unfortunately yes - its kinda against the rules rules to use rules from other rules....
I feel like breaking such a thing for temporary folders is not worth it...if it would be some bigger more expensive and configurable object I would possibly try to work on it more.

we could have it in a utility class or something - maybe it could be moved into the `ProjectPathUtils` the [other patch](https://github.com/apache/druid/pull/16249) introduces...not sure

Reply:
altered how this is being handled by:
* replaced `temporaryFolder` with `TempFolderProducer` 
* inlined these method to invoke `tempFolderProducer.newTempFolder`